redash/extensions.py

```diff
@@ -1,30 +1,50 @@
-import os
-from pkg_resources import iter_entry_points, resource_isdir, resource_listdir
+from importlib_metadata import entry_points
+
+# The global Redash extension registry
+extensions = {}
+
+# The periodic Celery task registry
+periodic_tasks = {}
 
 
 def init_extensions(app):
-    """
-    Load the Redash extensions for the given Redash Flask app.
-    """
-    if not hasattr(app, 'redash_extensions'):
-        app.redash_extensions = {}
+    """Load the Redash extensions for the given Redash Flask app.
 
-    for entry_point in iter_entry_points('redash.extensions'):
+    The extension entry point can return any type of value but
+    must take a Flask application object.
+
+    E.g.::
+
+        def extension(app):
+            app.logger.info("Loading the Foobar extenions")
+            Foobar(app)
+
+    """
+    for entry_point in entry_points().get('redash.extensions', []):
         app.logger.info('Loading Redash extension %s.', entry_point.name)
-        try:
-            extension = entry_point.load()
-            app.redash_extensions[entry_point.name] = {
-                "entry_function": extension(app),
-                "resources_list": []
+        init_extension = entry_point.load()
+        extensions[entry_point.name] = init_extension(app)
+
+
+def init_periodic_tasks(app):
```

Comment:
This is never called?

Reply:
Yeah, I first wanted to get a nod-off from you on the changed API, this would be called from the create_app function, too.

Reply:
It can't be called from `create_app`, because create_app is called after the point periodic tasks should be configured. This is something that should be explicitly called in `redash.worker`.

Reply:
Oh right, that's what I meant. Sorry